mycity/mycity/intents/grocery_store_intent.py

```diff
@@ -0,0 +1,137 @@
+    """
+    mycity_response = MyCityResponseDataModel()
+
+    # Get current address location
+    if CURRENT_ADDRESS_KEY in mycity_request.session_attributes:
+        current_address = \
+            mycity_request.session_attributes[CURRENT_ADDRESS_KEY]
+
+        # Parsing street address using street-address package
+        address_parser = StreetAddressParser()
+        a = address_parser.parse(current_address)
+        address = str(a["house"]) + " " + str(a["street_name"]) + " " \
+                  + str(a["street_type"])
+
+        # Parsing zip code
+        zip_code = str(a["other"]).zfill(5) if a["other"] else None
+        zip_code_key = intent_constants.ZIP_CODE_KEY
+        if zip_code is None and zip_code_key in \
+                mycity_request.session_attributes:
+            zip_code = mycity_request.session_attributes[zip_code_key]
+
+        # Get user's address and get grocery stores
+        usr_addr = gis_utils.geocode_address(address)
+        nearby_grocery_stores = get_grocery_grocery_stores(usr_addr)
+
+        try:
+            if len(nearby_grocery_stores):
+                response = 'The following markets are located within a mile ' \
+                           'from you. '
+                response += add_response_text(nearby_grocery_stores)
+                mycity_response.output_speech = response
+            else:
+                response = 'I did not find any grocery stores within a mile ' \
+                           'from your address.'
+                mycity_response.output_speech = response
+
+        except InvalidAddressError:
+            address_string = address
+            if zip_code:
+                address_string = address_string + " with zip code {}"\
+                    .format(zip_code)
+            mycity_response.output_speech = \
+                speech_constants.ADDRESS_NOT_FOUND.format(address_string)
+            mycity_response.dialog_directive = "ElicitSlotGroceryStore"
+            mycity_response.reprompt_text = None
+            mycity_response.session_attributes = \
+                mycity_request.session_attributes
+            mycity_response.card_title = "Grocery Store"
+            mycity_request = clear_address_from_mycity_object(mycity_request)
+            mycity_response = clear_address_from_mycity_object(mycity_response)
+            return mycity_response
+
+        except BadAPIResponse:
+            mycity_response.output_speech = \
+                "Hmm something went wrong. Maybe try again?"
+
+        except MultipleAddressError:
+            mycity_response.output_speech = \
+                speech_constants.MULTIPLE_ADDRESS_ERROR.format(address)
+            mycity_response.dialog_directive = "ElicitSlotZipCode"
+
+    else:
+        logger.error("Error: Called grocery_store_intent with no address")
+        mycity_response.output_speech = "I didn't understand that address, " \
+                                        "please try again"
```

Comment:
This is really a coding failure, so I am not sure if asking for the address would actually fix this. I would have a generic something went wrong message, so the user doesn't keep trying the same intent.